core/app/modules/kit-library/data/repository.php

```diff
@@ -213,11 +226,15 @@ private function get_taxonomies_data( $force_api_request = false ) {
 	 * @return array
 	 */
 	private function transform_kit_api_response( $kit, $manifest = null ) {
-		$taxonomies = array_reduce( static::TAXONOMIES_KEYS, function ( $current, $key ) use ( $kit ) {
-			return array_merge( $current, array_map( function ( $taxonomy ) {
-				return $taxonomy->name;
-			}, $kit->{$key} ) );
-		}, [] );
+		$subscription_plan_tag = $this->subscription_plans->get( $kit->access_level );
+
+		$taxonomies = ( new Collection( static::TAXONOMIES_KEYS ) )
+			->reduce( function ( Collection $current, $key ) use ( $kit ) {
+				return $current->merge(
+					( new Collection( $kit->{$key} ) )->pluck( 'name' )
+				);
+			}, new Collection( [] ) )
+			->merge( [ $subscription_plan_tag ? $subscription_plan_tag : 'Free' ] );
```

Comment:
Explain, and maybe make it more readable, it's really messy and complicated in my opinion.

Reply:
Done